Userland/Applications/Dictionary/Dictionary.cpp

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2021, Robin Allen <r@foon.uk>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include <AK/StringBuilder.h>
+
+#include "Dictionary.h"
+
+namespace Dictionary {
+
+static char const* part_of_speech_name(char pos)
+{
+    switch (pos) {
+    case 'n':
+        return "noun";
+    case 'v':
+        return "verb";
+    case 's':
+    case 'a':
+        return "adjective";
+    case 'r':
+        return "adverb";
+    default:
+        return "misc";
+    }
+}
+
+static int size_of_index_entry(char const* index_entry)
+{
+    int n = 0;
+    for (int i = 0; i < 16; i++) {
+        if (index_entry[i])
+            n++;
+        else
+            break;
+    }
+    return n;
+}
+
+Dictionary::Dictionary(char const* filename)
+{
+    auto file_or_error = MappedFile::map(filename);
+    if (file_or_error.is_error()) {
+        m_num_words = 0;
+        m_dictionary_bytes = 0;
```

Comment:
Not sure where/if it shows up elsewhere, but please use `nullptr` for the null pointer constant.

Reply:
Done.